zstd/zstd.go

```diff
@@ -17,6 +17,9 @@ const Code = 4
 
 const DefaultCompressionLevel = 3
 
+// DefaultDecoderPoolMaxSize is the default max number of cached decoders for a decoderPool.
+const DefaultDecoderPoolMaxSize = 100
```

Comment:
Arbitrarily chosen. Picked a relatively small number since each `zstd.Decoder` will spin up multiple goroutines.

I've been benchmarking an app with loads >>100 QPS per pod (small EC2 instances) and haven't seen the pool max out yet.